python/ray/data/datasource/parquet_datasource.py

```diff
@@ -172,32 +168,32 @@ def _file_format(self) -> str:
 
 
 def _fetch_metadata_remotely(
-        pieces: List[bytes]
+        pieces: List["pyarrow._dataset.ParquetFileFragment"]
 ) -> List[ObjectRef["pyarrow.parquet.FileMetaData"]]:
+    from ray import cloudpickle
     remote_fetch_metadata = cached_remote_fn(
         _fetch_metadata_serialization_wrapper)
     metas = []
     parallelism = min(len(pieces) // PIECES_PER_META_FETCH, 100)
     meta_fetch_bar = ProgressBar("Metadata Fetch Progress", total=parallelism)
-    for pieces_ in np.array_split(pieces, parallelism):
-        if len(pieces_) == 0:
+    for pcs in np.array_split(pieces, parallelism):
+        if len(pcs) == 0:
             continue
-        metas.append(remote_fetch_metadata.remote(pieces_))
+        metas.append(remote_fetch_metadata.remote(cloudpickle.dumps(pcs)))
```

Comment:
It's a shame that we'll have to serialize all pieces twice when metadata is fetched remotely... Have you tested this on an S3 dataset that will parallelize metadata fetching (greater than 24 files)?

Reply:
I'm not generally worried about this, since the overhead should be small in the first place. Avoiding O(n^2) blowup is more important.